_includes/config-guide/custom-logger-handler-examples.md

```diff
@@ -0,0 +1,157 @@
+You can use one of the following approaches for logging into custom file:
+- setup custom log file in the `di.xml`
+- setup custom file in the custom logger handler class 
+
+## Setup custom log file in the `di.xml`
+
+1. Define custom log file in the virtual type
+
+```xml
+<virtualType name="Magento\Payment\Model\Method\VirtualDebug" type="Magento\Framework\Logger\Handler\Base">
+    <arguments>
+        <argument name="fileName" xsi:type="string">/var/log/payment.log</argument>
+    </arguments>
+</virtualType>
+```
+
+1. Specify handler which will be used
+
+```xml
+<virtualType name="Magento\Payment\Model\Method\VirtualLogger" type="Magento\Framework\Logger\Monolog">
+    <arguments>
+        <argument name="handlers" xsi:type="array">
+            <item name="debug" xsi:type="object">Magento\Payment\Model\Method\VirtualDebug</item>
+        </argument>
+    </arguments>
+</virtualType>
+```
+
+1. Inject class of virtual logger to `$this->logger`
+
+```xml
+<type name="Magento\Payment\Model\Method\Logger">
+    <arguments>
+        <argument name="logger" xsi:type="object">Magento\Payment\Model\Method\VirtualLogger</argument>
+    </arguments>
+</type>
+```
+
+1. Virtual class `Magento\Payment\Model\Method\VirtualLogger` will be injected into `$this->logger` and will log exception messages into `/var/log/payment.log` file. 
+
+## Setup custom file in the custom logger handler class
+
+1. Create a class which will log the necessary data. 
+
+Create a new file `app/code/Vendor/ModuleName/Logger/Handler/ErrorHandler.php` with the next content
+
+```php
+<?php
+/**
+ * @author Vendor
+ * @copyright Copyright (c) 2019 Vendor (https://www.vendor.com/)
+ */
+namespace Vendor\ModuleName\Logger\Handler;
+
+use Magento\Framework\Logger\Handler\Base as BaseHandler;
+use Monolog\Logger as MonologLogger;
+
+/**
+ * Class ErrorHandler
+ */
+class ErrorHandler extends BaseHandler
+{
+    /**
+     * Logging level
+     *
+     * @var int
+     */
+    protected $loggerType = MonologLogger::ERROR;
+
+    /**
+     * File name
+     *
+     * @var string
+     */
+    protected $fileName = '/var/log/my_custom_logger/error.log';
+}
+```
+
+1. Define it for the specific handler type in the `di.xml`: 
+
+For example, `Vendor\ModuleName\Logger\Handler\ErrorHandler` will be defined for the error handler.
+
+```xml    
+<virtualType name="MyCustomLogger" type="Magento\Framework\Logger\Monolog">
```

Comment:
In the previous example we have named our virtual type as `Magento\Payment\Model\Method\VirtualLogger` here we name it as `MyCustomLogger`. For a reader who is not familiar with the virtual types it might seem like something important. But technically, it's just a unique name. I suggest using the same naming convention for all examples.

Reply:
**fixed**